src/systems/dom-overlay.js

```diff
@@ -0,0 +1,21 @@
+var registerSystem = require('../core/system').registerSystem;
+
+/**
+ * WebXR DOM Overlay support.
+ */
+module.exports.System = registerSystem('dom-overlay', {
+  schema: {
+    rootElement: {default: document.body, type: 'selector'},
+    required: {default: false, type: 'boolean'}
```

Comment:
When would someone pick required or optional?

Reply:
An application would use "required" if it's essentially unusable without the DOM Overlay, for example if the main information it displays is in the DOM, or if critical user interface elements only exist as DOM elements. 

It would use "optional" if it still works on a device where the overlay isn't supported, either because the functionality is secondary (i.e. a convenience "Exit AR" button), or because it has alternate input or display methods that don't need it. For example, the RC car sample I've been experimenting with uses the DOM overlay to show a screen controller, but an AR headset wouldn't need it since it uses controller input instead.

Reply:
There's also a middle ground where functionality may be degraded without the overlay, but it technically still works. It's up to application developers how to handle that, but my general recommendation would be to request it as optional, and consider showing a warning about missing functionality if it's not available and users may be confused otherwise.

Reply:
Thanks, wondering if `required` should be exposed on the schema or hide it for now and default to `optional` until we see a need to expose.